### PR TITLE
Point that route parameters are also Request attributes

### PR DESCRIPTION
**Pull request by sfdumi**
Point that route parameters are also Request attributes

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@sfdumi :+1: I like your proposal because a lot of people are not aware of this. Thank you!

File: cookbook/routing/extra_information.rst [hunk 1]
```diff
@@ -52,12 +52,16 @@ arguments to your controller:
...
-Now, you can access this extra parameter in your controller::
+Now, you can access this extra parameter in your controller, either as an argument (if specified), or through the ``Request`` object::
 
-    public function indexAction($page, $title)
+    use Symfony\Component\HttpFoundation\Request;
+    
+    public function indexAction(Request $request, $page, $title)
     {
         // ...
+        $titleAttribute = $request->attributes->get('title'); // same as $title
```
Comment (OskarStark) [inline cookbook/routing/extra_information.rst]: `$titleAttribute` is confusing imo, to show this, its enough to write `$request->attributes->get('title'); // same as $title`

what do you thing @javiereguiluz

Comment (javiereguiluz) [inline cookbook/routing/extra_information.rst]: I agree with you. We could even split the example in two parts:

```php
// Getting the title via controller arguments

public function indexAction($page, $title)
{
    // ...
}

// Getting the title via request attributes

use Symfony\Component\HttpFoundation\Request;

public function indexAction(Request $request)
{
    $title = $request->attributes->get('title');
    // ...
}
```

**Comment by sfdumi** [comment]
This sounds like a good idea. Should I update the PR?

File: cookbook/routing/extra_information.rst [hunk 2]
```diff
@@ -7,7 +7,7 @@ How to Pass Extra Information from a Route to a Controller
 Parameters inside the ``defaults`` collection don't necessarily have to
 match a placeholder in the route ``path``. In fact, you can use the
 ``defaults`` array to specify extra parameters that will then be accessible as
-arguments to your controller:
+arguments to your controller, and as attributes of the ``Request`` object:
```
Comment (xabbuh) [inline cookbook/routing/extra_information.rst]: We change this a bit and only tell the reader that we can access the parameters in the controller:

> [...] that will then be accessible in your controller::

File: cookbook/routing/extra_information.rst [hunk 3]
```diff
@@ -52,12 +52,16 @@ arguments to your controller:
 
         return $collection;
 
-Now, you can access this extra parameter in your controller::
+Now, you can access this extra parameter in your controller, either as an argument (if specified), or through the ``Request`` object::
```
Comment (xabbuh) [inline cookbook/routing/extra_information.rst]: I would remove the addition here and change the code to the example proposed by @javiereguiluz.

File: cookbook/routing/extra_information.rst [hunk 4]
```diff
@@ -52,12 +52,16 @@ arguments to your controller:
...
+    public function indexAction(Request $request, $page, $title)
     {
         // ...
+        $titleAttribute = $request->attributes->get('title'); // same as $title
+        // ...
     }
 
 As you can see, the ``$title`` variable was never defined inside the route path,
-but you can still access its value from inside your controller.
+but you can still access its value from inside your controller, or from the ``Request`` object's ``attributes`` bag.
```
Comment (xabbuh) [inline cookbook/routing/extra_information.rst]: Can you please break the line here after the first word that crosses the 72nd character?

**Comment by WouterJ** [comment]
I like this addition! If you can and agree with them, please update this PR to add the proposed changes.

**Comment by sfdumi** [comment]
Updated the PR.

**Comment by javiereguiluz** [comment]
:+1: this PR looks ready to be merged. Our doc managers can fix some long lines when merging this.

Thanks @sfdumi.

**Comment by WouterJ** [comment]
:+1: (I'm very sorry, the update comment completely slipped under my radar)

**Comment by xabbuh** [comment]
Thank you for these improvements @sfdumi! I have merged them into the `2.3` branch.

**Pull request closed without merging**